File: src/main/java/dev/thomas/polinizamap/entity/Avistamento.java
package dev.thomas.polinizamap.entity;

import dev.thomas.polinizamap.enums.StatusAvistamento;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "avistamentos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Avistamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "especie_id", nullable = false)
    private Especie especie;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "regiao_id", nullable = false)
    private Regiao regiao;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Column(columnDefinition = "TEXT")
    private String descricao;

    @Column(name = "data_hora", nullable = false)
    private LocalDateTime dataHora;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private StatusAvistamento status = StatusAvistamento.PENDENTE;

    @Column(name = "nota_validacao", columnDefinition = "TEXT")
    private String notaValidacao;
}

File: frontend/src/types/avistamento.ts
export type StatusAvistamento = 'PENDENTE' | 'APROVADO' | 'REJEITADO'

export interface AvistamentoRequest {
  especieId: number
  regiaoId: number
  latitude: number
  longitude: number
  descricao: string
  dataHora: string
}

export interface AvistamentoResponse {
  id: number
  usuarioNome: string
  especieNomePopular: string
  regiaoNome: string
  regiaoCidade: string
  latitude: number
  longitude: number
  descricao: string | null
  dataHora: string
  status: StatusAvistamento
  notaValidacao: string | null
}

export interface ValidacaoRequest {
  status: StatusAvistamento
  notaValidacao?: string | null
}
